### PR TITLE
Internalize some Docker types

### PR DESCRIPTION
**Pull request by nalind**
Internalize the type declaration of a Docker image, so that we don't need to pull in code and a pile of additional dependencies when we just want to know how to parse out a JSON object.  Carry a copy of the license that covered the sources from which the definitions came.

### CODE REVIEW

File: docker/LICENSE [hunk 1]
```diff
@@ -0,0 +1,191 @@
+
```
Comment (rhatdan) [inline docker/LICENSE]: Do we actually need this since Buildah is Apache licensed?

Comment (nalind) [inline docker/LICENSE]: Fair point.  I don't see us wanting to change the license, so I can drop that file.

**Comment by rhatdan** [comment]
LGTM

**Comment by nalind** [comment]
@rh-atomic-bot r+ fbf1b83adee1a70e67105075e316dfa57b8e2a6e

**Pull request closed without merging**